src/librbd/DiffIterate.cc

```diff
@@ -268,7 +268,8 @@ int DiffIterate::execute() {
     r = 0;
     if (m_image_ctx.parent && overlap > 0) {
       ldout(cct, 10) << " first getting parent diff" << dendl;
-      DiffIterate diff_parent(*m_image_ctx.parent, NULL, 0, overlap,
+      DiffIterate diff_parent(*m_image_ctx.parent, cls::rbd::UnknownSnapshotNamespace(),
```

Comment:
I presume in order to use this I should reorder Namespaces in the definition of Variant. So that UnknownSnapshotNamespace is the first one.